File: app/router/cloud.go
package router

import (
	"github.com/aihop/gopanel/app/api"
	"github.com/aihop/gopanel/app/middleware"
	"github.com/aihop/gopanel/constant"
	"github.com/gofiber/fiber/v3"
)

func CloudRouter(r fiber.Router) {
	cloudGroup := r.Group("/cloud")
	cloudGroup.Use(middleware.JWT(constant.UserRoleAdmin))
	{
		cloudGroup.Post("/account/search", api.CloudAccountSearch)
		cloudGroup.Post("/account", api.CloudAccountCreate)
		cloudGroup.Post("/account/update", api.CloudAccountUpdate)
		cloudGroup.Post("/account/del", api.CloudAccountDelete)
		cloudGroup.Get("/cdn/:id/domains", api.CloudCdnDomains)
	}
}

File: admin/src/api/modules/cloud.ts
import http from "@/api"
import type { ReqPage, ResPage } from "../interface"
import type { Website } from "../interface/website"

import { TimeoutEnum } from "@/enums/http-enum"

export const SearchCloudAccount = (req: ReqPage) => {
	return http.post<ResPage<Website.CloudAccount>>(`/cloud/account/search`, req)
}

export const CreateCloudAccount = (req: Website.CloudAccountCreate) => {
	return http.post<any>(`/cloud/account`, req)
}

export const UpdateCloudAccount = (req: Website.CloudAccountUpdate) => {
	return http.post<any>(`/cloud/account/update`, req)
}

export const DeleteCloudAccount = (req: Website.DelReq) => {
	return http.post<any>(`/cloud/account/del`, req)
}

export const CloudCdnDomainsAPI = (id: number) => {
	return http.get<string[]>(`/cloud/cdn/${id}/domains`)
}